sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -402,8 +397,10 @@ export class MessageSender extends LinkEntity {
       this.openLock
     );
 
-    return await defaultLock.acquire(this.openLock, async () => {
+    return defaultLock.acquire(this.openLock, async () => {
```

Comment:
Nice! And I assume you left `open` as an async function so you wouldn't have to return a `Promise.reject(AbortError)` in your check before acquiring a lock, instead of just throwing, right?